x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/AutoscalingIT.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.integration;
+
+import org.elasticsearch.action.admin.cluster.node.info.NodeInfo;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.xpack.autoscaling.Autoscaling;
+import org.elasticsearch.xpack.autoscaling.action.GetAutoscalingCapacityAction;
+import org.elasticsearch.xpack.autoscaling.action.PutAutoscalingPolicyAction;
+import org.elasticsearch.xpack.autoscaling.capacity.AutoscalingDeciderResult;
+import org.elasticsearch.xpack.autoscaling.capacity.AutoscalingDeciderResults;
+import org.elasticsearch.xpack.core.ml.job.config.AnalysisConfig;
+import org.elasticsearch.xpack.core.ml.job.config.AnalysisLimits;
+import org.elasticsearch.xpack.core.ml.job.config.DataDescription;
+import org.elasticsearch.xpack.core.ml.job.config.Detector;
+import org.elasticsearch.xpack.core.ml.job.config.Job;
+import org.elasticsearch.xpack.ml.MachineLearning;
+import org.elasticsearch.xpack.ml.autoscaling.MlAutoscalingDeciderService;
+import org.elasticsearch.xpack.ml.autoscaling.NativeMemoryCapacity;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+
+public class AutoscalingIT extends MlNativeAutodetectIntegTestCase {
+
+    private static final long BASIC_REQUIREMENT_MB = 10;
+    private static final long NATIVE_PROCESS_OVERHEAD_MB = 30;
+    private static final long BASELINE_OVERHEAD_MB = BASIC_REQUIREMENT_MB + NATIVE_PROCESS_OVERHEAD_MB;
+
+    // This test assumes that xpack.ml.max_machine_memory_percent is 30
+    // and that xpack.ml.use_auto_machine_memory_percent is false
+    public void testMLAutoscalingCapacity() {
+        SortedMap<String, Settings> deciders = new TreeMap<>();
+        deciders.put(MlAutoscalingDeciderService.NAME,
+            Settings.builder().put(MlAutoscalingDeciderService.DOWN_SCALE_DELAY.getKey(), TimeValue.ZERO).build());
+        final PutAutoscalingPolicyAction.Request request = new PutAutoscalingPolicyAction.Request(
+            "ml_test",
+            new TreeSet<>(),
```

Comment:
@benwtrent do you know why this did not work? It is preventing me from getting #66082 merged.

Reply:
@henningandersen when I tried to supply `"ml"` it complained saying it was not really a valid node role. Now, this may have been due to the ML plugin not being loaded correctly? The easiest way to check is to simply put `ml` in there in 7.x and ry to run the test. I will double check

Reply:
Yeah, I did that and get the validation error. I only wanted to be sure I was not chasing something you already investigated deeply.

Reply:
@henningandersen 
```
 REPRODUCE WITH: ./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.ml.integration.AutoscalingIT.testMLAutoscalingCapacity" -Dtests.seed=93A328B876AF5108 -Dtests.security.manager=true -Dtests.locale=pl -Dtests.timezone=Atlantic/Faroe -Druntime.java=11
  2> org.elasticsearch.action.ActionRequestValidationException: Validation Failed: 1: ml;
        at __randomizedtesting.SeedInfo.seed([93A328B876AF5108:D856B3907EB37506]:0)
        at org.elasticsearch.xpack.autoscaling.action.PutAutoscalingPolicyAction$Request.validate(PutAutoscalingPolicyAction.java:150)
        at org.elasticsearch.action.TransportActionNodeProxy.execute(TransportActionNodeProxy.java:42)
        at org.elasticsearch.client.transport.TransportProxyClient.lambda$execute$0(TransportProxyClient.java:55)
        at org.elasticsearch.client.transport.TransportClientNodesService.execute(TransportClientNodesService.java:253)
        at org.elasticsearch.client.transport.TransportProxyClient.execute(TransportProxyClient.java:55)
        at org.elasticsearch.client.transport.TransportClient.doExecute(TransportClient.java:391)
        at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:412)
        at org.elasticsearch.client.FilterClient.doExecute(FilterClient.java:65)
        at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:412)
        at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:401)
        at org.elasticsearch.xpack.ml.integration.AutoscalingIT.testMLAutoscalingCapacity(AutoscalingIT.java:57)
```

I **THINK** the node role format has changed in master vs 7.x. Passing `ml` as the only role works fine in master.

Reply:
Ahhhh, the validation runs in the transport client and this causes the issue, since the client does not know about the ml plugin. Using a core role like `master` passes that validation (but fails the test with my PR due to validation).

Reply:
Let me open a PR to move the validation out of the request in 7.x.